src/AsmResolver.Workspaces/Workspace.cs

```diff
@@ -25,7 +25,7 @@ public WorkspaceIndex Index
         /// Performs the analysis.
         /// </summary>
         /// <param name="context">The analysis context.</param>
-        protected void Analyze(AnalysisContext context)
+        public void Analyze(AnalysisContext context)
```

Comment:
This is not meant to be public. It would allow for arbitrary `AnalysisContext`s being passed on that are not associated with the workspace. `AnalysisContext` is only supposed to be instantiated by the workspace itself (which is also why the workspace is a parameter of the `AnalysisContext` constructor).